src/ssl/PeekingPeerConnector.cc

```diff
@@ -84,7 +84,7 @@ Ssl::PeekingPeerConnector::checkForPeekAndSplice()
     if (!srvBio->canBump())
         acl_checklist->banAction(Acl::Answer(ACCESS_ALLOWED, Ssl::bumpBump));
     acl_checklist->syncAle(request.getRaw(), nullptr);
-    acl_checklist->nonBlockingCheck(Ssl::PeekingPeerConnector::cbCheckForPeekAndSpliceDone, this);
+    acl_checklist.release()->nonBlockingCheck(Ssl::PeekingPeerConnector::cbCheckForPeekAndSpliceDone, this);
```

Comment:
I fear the proposed solution because this code line will also compile _without_ `.release()` addition. In other words, the above line change is not necessary to compile Squid after the acl_checklist initialization was changed to use unique_ptr returned by Make(), but compiled Squid may crash.

To solve this problem, I think we have to make nonBlockingCheck() private and provide a public static NonBlockingCheck() method that accepts unique_ptr reference as a parameter. If you do not have a better idea, let's implement this one.

Reply:
I applied PR258 changes (769e66b) and adjusted accordingly (e252105).